src/frontends/common/src/complex_type_mark.cpp

```diff
@@ -289,6 +291,52 @@ ov::Output<ov::Node> ComplexTypeMark::div(const NodeContext& context,
     return {result};
 }
 
+ov::Output<ov::Node> ComplexTypeMark::equal(const NodeContext& context,
+    const ov::Output<ov::Node>& lhs,
+    const ov::Output<ov::Node>& rhs) {
+auto lhs_complex = as_type_ptr<ComplexTypeMark>(lhs.get_node_shared_ptr());
+auto rhs_complex = as_type_ptr<ComplexTypeMark>(rhs.get_node_shared_ptr());
+
+if (lhs_complex && rhs_complex) {
+// both operands are of complex type
+auto lr = lhs_complex->get_real();
+auto li = lhs_complex->get_imag();
+auto rr = rhs_complex->get_real();
+auto ri = rhs_complex->get_imag();
```

Comment:
please think how to implement it using `get_data()` and how it will simplify this decomposition